engine/common/synchronization/request_handler.go

```diff
@@ -0,0 +1,370 @@
+package synchronization
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/messages"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/lifecycle"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/network"
+	"github.com/onflow/flow-go/storage"
+)
+
+// defaultSyncRequestQueueCapacity maximum capacity of sync requests queue
+const defaultSyncRequestQueueCapacity = 500
+
+// defaultSyncRequestQueueCapacity maximum capacity of range requests queue
+const defaultRangeRequestQueueCapacity = 500
+
+// defaultSyncRequestQueueCapacity maximum capacity of batch requests queue
+const defaultBatchRequestQueueCapacity = 500
+
+// defaultEngineRequestsWorkers number of workers to dispatch events for requests
+const defaultEngineRequestsWorkers = 8
+
+type RequestHandler struct {
```

Comment:
`RequestHandlerEngine` and `request_handler_engine.go` since it is a fully qualified engine, maybe?

Reply:
That sounds good to me.